cashu/lightning/lnd_grpc/lnd_grpc.py

```diff
@@ -0,0 +1,376 @@
+import asyncio
+import codecs
+import hashlib
+import os
+from typing import AsyncGenerator, Optional
+
+import bolt11
+import grpc
+from bolt11 import (
+    TagChar,
+)
+from grpc.aio import AioRpcError
+from loguru import logger
+
+import cashu.lightning.lnd_grpc.protos.lightning_pb2 as lnrpc
+import cashu.lightning.lnd_grpc.protos.lightning_pb2_grpc as lightningstub
+import cashu.lightning.lnd_grpc.protos.router_pb2 as routerrpc
+import cashu.lightning.lnd_grpc.protos.router_pb2_grpc as routerstub
+from cashu.core.base import Amount, MeltQuote, Unit
+from cashu.core.helpers import fee_reserve
+from cashu.core.settings import settings
+from cashu.lightning.base import (
+    InvoiceResponse,
+    LightningBackend,
+    PaymentQuoteResponse,
+    PaymentResponse,
+    PaymentStatus,
+    PostMeltQuoteRequest,
+    StatusResponse,
+)
+
+
+class LndRPCWallet(LightningBackend):
+
+    supports_mpp = settings.mint_lnd_enable_mpp
+    supports_incoming_payment_stream = True
+    supported_units = set([Unit.sat, Unit.msat])
+    unit = Unit.sat
+
+    def __init__(self, unit: Unit = Unit.sat, **kwargs):
+        self.assert_unit_supported(unit)
+        self.unit = unit
+        self.endpoint = settings.mint_lnd_rpc_endpoint
+        cert_path = settings.mint_lnd_rpc_cert
+
+        macaroon_path = (
+            settings.mint_lnd_rpc_macaroon
+            or settings.mint_lnd_rpc_admin_macaroon
+            or settings.mint_lnd_rpc_invoice_macaroon
```

Comment:
I think we should remove the options for other macaroons since a macaroon with sufficient abilities will always be required to run the mint